### PR TITLE
FIX: ClassCastException when commonHandle with type Timestamp and Date

### PR DESCRIPTION
**Pull request by xuanyu66**
### What problem does this PR solve? 

Fix ClassCastException when commonHandle with type Timestamp and Date.

### What is changed and how it works?

https://github.com/pingcap/tispark/issues/2318.
First, use getTime() to get milliseconds and then convert to the desired type.

### Check List 

Tests 

 - Integration test

Related changes

 - Need to cherry-pick to the release-2.5, release-2.4
 - Need to be included in the release note

### CODE REVIEW

**Review by birdstorm — APPROVED** [review]
LGTM

**Review by shiyuhang0 — APPROVED** [review]

**Comment by shiyuhang0** [comment]
@xuanyu66 please fix the fail IT

**Review by shiyuhang0 — CHANGES_REQUESTED** [review]
@xuanyu66 please fix the fail IT

File: tikv-client/src/main/java/com/pingcap/tikv/key/CommonHandle.java [hunk 1]
```diff
@@ -53,11 +53,20 @@ public static CommonHandle newCommonHandle(
     CodecDataOutput cdo = new CodecDataOutput();
     for (int i = 0; i < data.length; i++) {
       if (dataTypes[i].getType().equals(MySQLType.TypeTimestamp)) {
-        long milliseconds = ((Timestamp) data[i]).getTime();
+        // When writing `Timestamp`, it will pass `Timestamp` object.
+        // When indexScan or tableScan, it will pass `Long` object.
+        // It's a compromise here since we don't have a good way to make them consistent.
+        long milliseconds;
```
Comment (shiyuhang0) [inline tikv-client/src/main/java/com/pingcap/tikv/key/CommonHandle.java:59]: use pattern match may be better
- for Timestamp use getTime() when
- for Long /1000
- for other type throw exception

**Pull request merged**